### ISSUE TITLE
Quality loss

### ISSUE THREAD

**Issue by jdanthinne** [kosso/TiAdobeCreativeSDK#3]
Is it normal that the resulting image size (in pixels) is smaller than the original Blob? Ie. an image originally at 1600x1200 results in a 1024x768 one. It's a big quality loss. 
Is it due to the module or the SDK itself?

**Comment by kosso** [comment]
Hi,
This implementation of the Adobe Image Editor SDK is only using the basic output options. (The earlier Aviary version of this SDK had a high resolution output as part of a 'premium' upgrade, which I never used).  This new version apparently allows high resolution output for free.  I just haven't figured out how to implement it yet. 

There's some documentation on 'High Resolution Output' [here]((https://creativesdk.adobe.com/docs/ios/#/articles/imageediting/index.html)  


I'll see what I can figure out, but it anyone else can help, I'm happy to take any pull requests! :)

**Comment by kosso** [comment]
OK. I just had a go with a modified version of the module, using a large photo 2592x1944 : 4.03 Mb

The rendering took a *long* time (and took a *lot* of memory). Also, the resulting edited image was the same dimensions 2592x1944, but the file size was 10.63 Mb !!?

I do see there is a way to provide a 'progress' while rendering the image, which would be required for a decent user experience, but I still think the image would need to be compressed before using it to upload somewhere to an online service. 

I'll have more of a play with the code when I get the time.

**Comment by jdanthinne** [comment]
Yes, I see in the "Enqueuing High Resolution Renders" that it is possible to define maximum dimensions to return. It would be great to be able to define that when requesting the editor.
I'm not very fluent in Objective-c but I'll have a look as well.

**Comment by jdanthinne** [comment]
Tried some code but Objective-C is definitely too strange for me… But I've found that https://github.com/ghkim/AviaryModule has a method that handles hires output. Perhaps you could get some inspiration there…

**Comment by kosso** [comment]
Yeah. As I said, I've tried it, and it works, but the output image is HUGE. (It could be compressed with another module though).

My issue is that I need to learn how to do the 'progress' while it processes the hires image. It can take a few seconds! *so much memory!*

**Comment by jdanthinne** [comment]
I understand. The ideal thing should be to define *hires* (bool) and *maxdims* (dic) options… If not *hires*, use the function as before, and if *hires*, use the *maxdims*, and let the user uses the image at its own risks.

For the compression, I was already thinking of https://github.com/appcelerator-modules/ti.imagefactory, which can compress images. But I think it's up the the dev to think about compression. Your module should just return the image as is.

**Comment by jdanthinne** [comment]
Hi, have you made any progress on this?
